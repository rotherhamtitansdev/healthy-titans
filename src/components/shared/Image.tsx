import React from "react";


// eslint-disable-next-line react/require-default-props
const Image = (props: {img: string; alt: string; additionalStyling?: string | undefined;}) => (
    <div className="px-[2.75rem] md:px-[5rem] pb-[0.5rem] md:pb-[1rem] min-w-[89px] min-h-[81px] 2xl:mt-5">
      <img
        className={`mx-auto ${props.additionalStyling}`}
        src={props.img}
        alt={props.alt}
        draggable={false}
      />
    </div>
  );

export default Image;
