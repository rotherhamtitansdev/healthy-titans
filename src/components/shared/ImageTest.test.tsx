import React from "react";
import { render, screen } from "@testing-library/react";
import Image from "./Image";

describe("Image Component", () => {
  test("renders image with correct alt and name", () => {
    const testAlt = "testAlt";
    const testURI = "testURI.svg";
    const { asFragment } = render(<Image img={testURI} alt={testAlt} />);

    expect(screen.getByRole("img", { name: testAlt })).toHaveAttribute("src", testURI);
    expect(asFragment()).toMatchSnapshot();
  });
});
