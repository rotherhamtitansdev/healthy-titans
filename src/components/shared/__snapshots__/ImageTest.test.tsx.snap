// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component renders image with correct alt and name 1`] = `
<DocumentFragment>
  <div
    class="px-[2.75rem] md:px-[5rem] pb-[0.5rem] md:pb-[1rem] min-w-[89px] min-h-[81px] 2xl:mt-5"
  >
    <img
      alt="testAlt"
      class="mx-auto undefined"
      draggable="false"
      src="testURI.svg"
    />
  </div>
</DocumentFragment>
`;
